src/Framework/PaymentGateways/Routes/GatewayRoute.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Give\Framework\PaymentGateways\Routes;
+
+use Give\Framework\PaymentGateways\Contracts\OffsiteGatewayInterface;
+use Give\Framework\PaymentGateways\DataTransferObjects\GatewayRouteData;
+use Give\Framework\PaymentGateways\Exceptions\PaymentGatewayException;
+use Give\Framework\PaymentGateways\PaymentGateway;
+use Give\Framework\PaymentGateways\PaymentGatewayRegister;
+use Give\Framework\PaymentGateways\Types\OffSitePaymentGateway;
+
+/**
+ * @unreleased
+ */
+class GatewayRoute
+{
+    /**
+     * Register route with a matching gateway method
+     *
+     * @return void
+     * @throws PaymentGatewayException
+     */
+    public function __invoke()
+    {
+        if ($this->isValidListener()) {
+            $paymentGateways = give(PaymentGatewayRegister::class)->getPaymentGateways();
+            $gatewayIds = array_keys($paymentGateways);
+
+            if (!$this->isValidRequest($gatewayIds)) {
+                throw new PaymentGatewayException('This route is not valid.');
+            }
+
+            $data = GatewayRouteData::fromRequest($_GET);
+
+            /** @var PaymentGateway $gateway */
+            $gateway = give($paymentGateways[$data->gatewayId]);
+
+            $gatewayMethods = array_filter($gateway::routeMethods, static function ($method) use ($data) {
+                return $method === $data->gatewayMethod;
+            });
+
+            if (is_a($gateway, OffsiteGatewayInterface::class)) {
+                $gatewayMethods = array_merge($gatewayMethods, OffSitePaymentGateway::routeMethods);
```

Comment:
@jonwaldstein Remember that the class is for convenience. Under the hood the gateways or compositional. As such, let's add `routeMethods` to the interface instead of the class.

Reply:
@JasonTheAdams I tried that first but was getting an error _Cannot inherit previously-inherited or override constant from interface_

Reply:
Interesting. We could try either static properties (my only issue with this is that the property becomes externally writeable), or just give the interface unique constant names. I'm fine with this as they're really only intended for internal use.